loader/src/main/scala/hmda/loader/lar/HmdaBatchLarLoader.scala

```diff
@@ -0,0 +1,140 @@
+package hmda.loader.lar
+
+import java.io.File
+import java.time.Instant
+
+import akka.pattern.ask
+import akka.actor.{ ActorPath, ActorRef, ActorSystem }
+import akka.cluster.client.{ ClusterClient, ClusterClientSettings }
+import akka.stream.{ ActorMaterializer, IOResult }
+import akka.stream.scaladsl.{ FileIO, Sink, Source }
+import akka.util.{ ByteString, Timeout }
+import hmda.api.util.FlowUtils
+import hmda.model.fi.{ Created, Filing, Submission }
+import hmda.persistence.HmdaSupervisor.{ FindFilings, FindHmdaFiling, FindProcessingActor, FindSubmissions }
+import hmda.persistence.institutions.FilingPersistence.CreateFiling
+import hmda.persistence.institutions.SubmissionPersistence.CreateSubmission
+import hmda.persistence.institutions.{ FilingPersistence, SubmissionPersistence }
+import hmda.persistence.processing.HmdaRawFile.AddLine
+import hmda.persistence.processing.ProcessingMessages.{ CompleteUpload, Persisted, StartUpload }
+import hmda.persistence.processing.SubmissionManager
+import hmda.persistence.messages.CommonMessages._
+import hmda.persistence.processing.SubmissionManager.AddFileName
+import hmda.query.HmdaQuerySupervisor.FindHmdaFilingView
+import org.slf4j.LoggerFactory
+
+import scala.concurrent.duration._
+import scala.concurrent.{ ExecutionContext, Future }
+import scala.util.{ Failure, Success }
+
+object HmdaBatchLarLoader extends FlowUtils {
+
+  override implicit val system: ActorSystem = ActorSystem("hmda-cluster-client")
+  override implicit val materializer: ActorMaterializer = ActorMaterializer()
+  override implicit val ec: ExecutionContext = system.dispatcher
+
+  val hmdaClusterName = config.getString("hmda.clusterName")
+  val hmdaClusterIP = config.getString("hmda.lar.host")
+  val hmdaClusterPort = config.getInt("hmda.lar.port")
+  val actorTimeout = config.getInt("hmda.actorTimeout")
+  val flowParallelism = config.getInt("hmda.lar.parallelism")
+
+  implicit val timeout = Timeout(actorTimeout.seconds)
+
+  val log = LoggerFactory.getLogger("hmda-lar-loader")
+
+  val initialContacts = Set(
+    ActorPath.fromString(s"akka.tcp://$hmdaClusterName@$hmdaClusterIP:$hmdaClusterPort/system/receptionist")
+  )
+
+  val settings = ClusterClientSettings(system)
+    .withInitialContacts(initialContacts)
+
+  val clusterClient = system.actorOf(ClusterClient.props(settings), "hmda-lar-loader")
+
+  def main(args: Array[String]): Unit = {
+
+    if (args.length < 2) {
+      exitSys(log, "Please provide a directory containing files with LAR data and period to process", 1)
+    }
+
+    val path = new File(args(0))
+    if (!path.isDirectory) {
+      exitSys(log, "Argument must be the full path to a directory containing files with LAR data", 2)
```

Comment:
When I smoke test, I get this error message every time, even with full file paths (starting with `/Users/`), that are definitely LAR files.

Reply:
When I switched this line to allow me to proceed, I get another error:

```
Association with remote system [akka.tcp://hmda@127.0.0.1:0] has failed, 
  address is now gated for [5000] ms. Reason: [Association failed with [akka.tcp://hmda@127.0.0.1:0]] 
  Caused by: [Can't assign requested address]
```

Maybe this is because I chose the wrong value for the `HMDA_CLUSTER_PORT` environment variable. If that's the case, where should I look to find the correct value?

Reply:
On the first error, you mention "full file paths", if you are passing individual files that might be the issue. It's expecting a directory (without trailing slash) with the files in it, followed by the period (i.e. 2016). In addition to this, the file names need to be in a specific format (the files we are using to test this are not public; ask about this offline). 

The association problem is most likely the port as you mention. You can do two things:
1) Set the application port explicitly with the `APP_PORT` environment variable when running the cluster, and use that same value in the `HMDA_CLUSTER_PORT` environment variable when running the `LAR` loader.
2) Look at the beginning of the logs when running the application, and taking note of the port that `Akka` assigns to the clustered application and assign it to the `HMDA_CLUSTER_PORT`. The 5th line or so after the HMDA Platform banner should have the information (i.e. _Remoting started; listening on addresses :[akka.tcp://hmda@127.0.0.1:**62096**]_). This is documented [here](https://github.com/jmarin/hmda-platform/blob/2016-lar-loader/Documents/2016_Data_load.md) as part of this pull request.